DB確認

```bash
ae2team2-db-1     | 2023-09-06T02:06:11.226652Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
ae2team2-db-1     | 2023-09-06T02:06:11.279557Z 1 [ERROR] [MY-012585] [InnoDB] Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
ae2team2-db-1     | 2023-09-06T02:06:11.280056Z 1 [Warning] [MY-012654] [InnoDB] Linux Native AIO disabled.
ae2team2-db-1     | 2023-09-06T02:06:11.962430Z 1 [ERROR] [MY-013171] [InnoDB] Cannot boot server version 80034 on data directory built by version 80100. Downgrade is not supported
ae2team2-db-1     | mysqld: 
ae2team2-db-1     | Can't open file: 'mysql.ibd' (errno: 0 - )
ae2team2-db-1     | 2023-09-06T02:06:11.977662Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine
ae2team2-db-1     | 2023-09-06T02:06:11.979212Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
ae2team2-db-1     | 2023-09-06T02:06:11.979463Z 0 [ERROR] [MY-010119] [Server] Aborting
ae2team2-db-1     | 2023-09-06T02:06:11.987819Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.34)  MySQL Community Server - GPL.
ae2team2-db-1 exited with code 1
```

## Linux Native AIO問題
```bash
InnoDB: Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
```
[command: mysqld --innodb_use_native_aio=0→解決](https://qiita.com/yoshiki_murakami/items/2306070624cfc31bcfdf)

## table flag問題
```bash
ae2team2-db-1     | 2023-09-06T05:38:41.384273Z 0 [ERROR] [FATAL] InnoDB: Table flags are 0 in the data dictionary but the flags in file ./ibdata1 are 0x4000!
```
[キャッシュが残ってるから→解決](https://yama-weblog.com/can-not-start-mysql-in-docker-when-version-down-grade/)
1. `docker/mysql/data`のフォルダを全て消す
2. `docker-compose up`し直す

## Cannot boot server version問題
```bash
ae2team2-db-1     | 2023-09-06T04:47:54.688176Z 1 [ERROR] [MY-013171] [InnoDB] Cannot boot server version 80034 on data directory built by version 80100. Downgrade is not supported
```
[前のcacheが残っているっぽい](https://qiita.com/deTeb/items/9c6bd1803ece3535fe74)
`docker-compose down -v`でボリュームを消す→解決せず
上記2つの問題に対処しているうちに消えました。

--- a/README.md
+++ b/README.md
@@ -49,5 +49,12 @@ docker-compose down
 docker-compose down --rmi all
 ```
 
+## DBのコンテナでエラーが出た場合
+```bash
+ae2team2-db-1     | 2023-09-06T05:38:41.384273Z 0 [ERROR] [FATAL] InnoDB: Table flags are 0 in the data dictionary but the flags in file ./ibdata1 are 0x4000!
+```
+1. `docker/mysql/data`のフォルダを全て消す
+2. `docker-compose up`し直す
+
 ## 参考
 https://github.com/shinjiezumi/vue-go-samples

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -43,6 +43,7 @@ services:
       MYSQL_DATABASE: sample
       MYSQL_ROOT_PASSWORD: root
       TZ: Asia/Tokyo
+    command: mysqld --innodb_use_native_aio=0 # M1 Mac
 #  docs:
 #    build:
 #      context: .

--- a/docker/mysql/Dockerfile
+++ b/docker/mysql/Dockerfile
@@ -1,5 +1,6 @@
-FROM mysql:8.0-debian
+# FROM mysql:8.0-debian
 # FROM mysql:5.7
+FROM mysql:5.7-debian
 
 # M1 Mac でapt-getできないのに対処
 # RUN sed -i s/deb.debian.org/archive.debian.org/g /etc/apt/sources.list
@@ -8,6 +9,6 @@ FROM mysql:8.0-debian
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
     vim
-RUN mkdir /var/log/mysql
+RUN mkdir -p /var/log/mysql
 RUN chown mysql:mysql /var/log/mysql
 RUN touch /var/log/mysql/mysqld.log

